### PR TITLE
[DONT MERGE] (PCP-209) Unvendor cpp-pcp-client and leatherman

### PR DESCRIPTION
**Pull request by parisiale**
Removing submodules.
Updating Travis and Appveyor configuration scripts.
Updating README.

### CODE REVIEW

**Comment by mruzicka** [comment]
LGTM.
@richardc can you please review it too?

**Comment by richardc** [comment]
Seems the README commit (6e1cd8143), though I like it, could come in as a separate concern as it's not really about any changes to the build system.

Why are we gaining a curl dependency on appveyor? I know leatherman has a curl wrapper but can't it be excluded as this code doesn't as far as I know use curl.

I'm not keen on the changes to the cmakelists.txt that are simply to make the comments read like sentences, as it introduces noise when trying to look at an annotated version of the code.

Is the reason these travis/appveyor builds failing just because this blocks on https://github.com/puppetlabs/cpp-pcp-client/pull/132 If so I'd like to be sure it builds once that's merged.

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,53 +41,45 @@ if ("${INSTALL_IS_SYSTEM_DIR}" STREQUAL "-1")
...
+include(${VENDOR_DIRECTORY}/horsewhisperer.cmake)
+
+# Pull in helper macros for working with leatherman libraries
+include(leatherman)
+
 # Leatherman it up
-list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/vendor/leatherman/cmake")
 include(options)
+include(cflags)
+option(BOOST_STATIC "Use Boost's static libraries" OFF)
```
Comment (richardc) [inline CMakeLists.txt]: Shouldn't this option be up around line 12 with the others?

**Comment by parisiale** [comment]
Thanks. I'll make another PR for the README.
I'll get back on this once we got https://github.com/puppetlabs/cpp-pcp-client/pull/132 in.

**Comment by parisiale** [comment]
Closing this since I created https://github.com/puppetlabs/pxp-agent/pull/299 without the README commit for references and without the `BOOST_STATIC` definition.

**Pull request closed without merging**